lib/FrontendTool/FrontendTool.cpp

```diff
@@ -592,26 +594,33 @@ static bool performCompile(std::unique_ptr<CompilerInstance> &Instance,
   if (Invocation.getSILOptions().LinkMode == SILOptions::LinkAll)
     performSILLinking(SM.get(), true);
 
-  {
-    SharedTimer timer("SIL verification (pre-optimization)");
-    SM->verify();
+  if (!PrimarySourceFile) {
+    SM->linkAll();
+    SM->dump();
```

Comment:
It's well understood that `-Onone` does not do what it says. The `-disable` flag should also disable the Onone optimizations. It's just that no one has bothered implementing it.